components/org.wso2.carbon.identity.sso.saml.cloud/src/main/java/org/wso2/carbon/identity/sso/saml/cloud/processor/AuthnRequestProcessor.java

```diff
@@ -99,4 +102,49 @@ public String getRelyingPartyId() {
         responseBuilder.setRedirectURL(commonAuthURL);
         return responseBuilder;
     }
+
+    protected SAMLCloudFrameworkLogoutResponse.SAMLCloudFrameworkLogoutResponseBuilder buildResponseForCloudLogout(
+            IdentityMessageContext context) {
+        IdentityRequest identityRequest = context.getRequest();
+        Map parameterMap = identityRequest.getParameterMap();
+        AuthenticationRequest authenticationRequest = new AuthenticationRequest();
+        authenticationRequest.appendRequestQueryParams(parameterMap);
+        for (Object entry : identityRequest.getHeaderMap().keySet()) {
+            authenticationRequest.addHeader((String) entry, identityRequest.getHeaderMap().get(entry));
+        }
+
+        authenticationRequest.setRelyingParty(((SAMLMessageContext) context).getIssuer());
+        authenticationRequest.setType(this.getName());
+
+        try {
+            authenticationRequest.setCommonAuthCallerPath(URLEncoder.encode(getCallbackPath(context),
+                                                                            StandardCharsets.UTF_8.name()));
+        } catch (UnsupportedEncodingException e) {
+            throw FrameworkRuntimeException.error("Error occurred while URL encoding callback path " +
+                                                  this.getCallbackPath(context), e);
+        }
+
+        authenticationRequest.addRequestQueryParam(FrameworkConstants.RequestParams.LOGOUT,
+                                                   new String[]{Boolean.TRUE.toString()});
+        AuthenticationRequestCacheEntry authRequest = new AuthenticationRequestCacheEntry(authenticationRequest);
+        String sessionId;
+        Cookie ssoTokenIdCookie = SAMLSSOUtil.getTokenIdCookie(context);
+        if (ssoTokenIdCookie != null) {
+            sessionId = ssoTokenIdCookie.getValue();
+        } else {
+            throw FrameworkRuntimeException.error("SSO Token ID cookie cannot be found");
```

Comment:
Cannot work without exception. We will need to do error handling in IdentityServlet. It has the facilities and in current solution also we haven't used them. So creating a pub jira.
https://wso2.org/jira/browse/IDNCLOUD-339